### ISSUE TITLE
Optimal hyperparameter settings

### ISSUE THREAD

**Issue by Jurio0304** [payalmohapatra/SilentSpeechLLM#1]
I'm trying to reproduce the results from your paper using this code, but I cannot get the optimal performance reported. Could you please share the optimal hyperparameter settings used to achieve the best results? That would be a huge help. Thanks!

Here are some hyperparameters I used and performance:

seed_num: 42
num_epochs: 1000
early_stop: 200
num_beams: 3
initial_lr: 1e-05
max_lr: 0.001
warmup_steps: 500
dropout: 0.2
batch_size: 8

---------------------------
Best train loss: 0.0110
Best train WER: 0.8403
Best validation loss: 1.3152
Best validation WER: 0.8750
Early stopping was triggered; the best model is at epoch 73.

**Comment by Jurio0304** [comment]
Hi again, just wanted to let you know that I’ve figured out the issue and successfully resolved it. I can now reproduce the optimal results as reported. Thanks again for your work!

**Comment by jinny1208** [comment]
Hi @Jurio0304 , do you mind telling us what the issue was and how you solved it?

**Comment by payalmohapatra** [comment]
> Hi [@Jurio0304](https://github.com/Jurio0304) , do you mind telling us what the issue was and how you solved it?

Hello @jinny1208 we have shared the checkpoint to reproduce our results in the readme as follows,  

> The best checkpoint can be downloaded from [here](https://drive.google.com/open?id=1xtq_UkSTn1rwMRt8KD_-Vz3chABIaJnR&usp=drive_copy).

**Comment by abdulloh3835** [comment]
> Hi again, just wanted to let you know that I’ve figured out the issue and successfully resolved it. I can now reproduce the optimal results as reported. Thanks again for your work!

Hi @Jurio0304, congratulations on reproducing the results. Could you please share what the specific fix was? Even one or two sentences on what made the biggest difference would help enormously. Thanks.
